subprojects/antlr/src/main/java/org/gradle/api/plugins/antlr/AntlrTask.java

```diff
@@ -152,6 +157,35 @@ public void setArguments(List<String> arguments) {
         return arguments;
     }
 
+    /**
+     * Pattern of regular expressions to match antlr include files.
+     *
+     * If antlr grammar files reference include files, specify regular expressions to match these include files.
+     * If include file patterns are specified, incremental grammar-build will be disabled.
+     * Include files will not be passed as grammar source files to the antlr tool.
+     *
+     * @return The include-file patterns.
+     * @since 6.8
+     */
+    @Input
+    @Incubating
+    public Set<Pattern> getIncludeFilesPatterns() {
+        return includeFilesPatterns;
+    }
+
+    /**
+     * Pattern of regular expressions to match antlr include files.
+     *
+     * See {@link #getIncludeFilesPatterns()}
+     *
+     * @param includeFilesPatterns The include-file patterns.
+     * @since 6.8
+     */
+    @Incubating
+    public void setIncludeFilesPatterns(Set<Pattern> includeFilesPatterns) {
+        this.includeFilesPatterns = includeFilesPatterns;
+    }
```

Comment:
Why pass a regular expression here. Gradle has a much more structured way of specifying file filters using `PatternFilterable` or `FileTreeElement`. I believe that this is already supported here.

https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceTask.html#org.gradle.api.tasks.SourceTask:include(org.gradle.api.specs.Spec)

What's missing from that API that you need to support here?